plugins.go

```diff
@@ -224,10 +225,24 @@ func (d *DynamicMiddleware) ProcessRequest(w http.ResponseWriter, r *http.Reques
 		"prefix": "jsvm",
 	}).Debug("JSVM middleware execution took: (ns) ", time.Now().UnixNano()-t1)
 
-	if newRequestData.Request.ReturnOverrides.ResponseCode != 0 {
+	if newRequestData.Request.ReturnOverrides.ResponseCode >= 400 {
 		return errors.New(newRequestData.Request.ReturnOverrides.ResponseError), newRequestData.Request.ReturnOverrides.ResponseCode
 	}
 
+	if newRequestData.Request.ReturnOverrides.ResponseCode != 0 && newRequestData.Request.ReturnOverrides.ResponseCode <= 300 {
```

Comment:
I still don't understand why we treat `300` differently from other 3xx codes here. Don't you mean `< 300`, to capture all 2xx and 1xx codes?

Reply:
Actually a good point, because our co-process functionality (which should be synced with this one, just use `> 0` https://github.com/TykTechnologies/tyk/blob/master/coprocess.go#L274

Reply:
My point was simply about the `3xx` inconsistency, but your point seems correct too. @lonelycode?